GNSS/src/GNSSManager.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef GNSS_MANAGER_H
+#define GNSS_MANAGER_H
+
+#include <Arduino.h>
+#include "rover_lib2/helpers/log.hpp"
+#include "rover_lib2/helpers/macros.hpp"
+#include "rover_lib2/sensors/gnss_parser.hpp"
+#include "rover_lib2/filters/circular_moving_average.hpp"
+
+struct sGNSSData
+{
+    float latitude = 0.0f;
+    float longitude = 0.0f;
+    Constants::eGGAQuality fixQuality = Constants::eGGAQuality::UNKNOWN;
+    uint8_t satellites = 0;
+    float headingDeg = 0.0f;
+    Constants::eUniHeadingQuality headingQuality = Constants::eUniHeadingQuality::NO_HEADING;
+
+    inline bool hasValidFix() const
+    {
+        return fixQuality != Constants::eGGAQuality::UNKNOWN;
+    }
+};
+
+/**
+ * @brief
+ * @attention Begin must be called on serial
+ *
+ */
+class GNSSManager
+{
+  private:
+    static constexpr size_t CIRCULAR_WINDOW_SIZE = 10;
+    static constexpr size_t MAX_SENTENCE_LENGTH = 200;
+
+    Stream& _GNSSSerial;
+    sGNSSData _currentData;
+    CircularMovingAverage<CIRCULAR_WINDOW_SIZE> _headingFilter;
+
+    char _sentenceBuffer[MAX_SENTENCE_LENGTH] = {'\0'};
+    size_t _buffer_Index;
+
+  public:
+    explicit GNSSManager(Stream& serial_);
+    void update(void);
+    sGNSSData getData(void);
+    float getFilteredHeading(void);
+
+  private:
+    void parseMSG(char* buffer, size_t length_);
```

Comment:
take std::array instead (safer since it passes length directly and checks at compile time)

```suggestion
template<size_t SIZE>
void parseMSG(std::array<char, SIZE> buffer_);
```